src/metabase/models/collection.clj

```diff
@@ -936,19 +935,36 @@
 ;;; |                                              Personal Collections                                              |
 ;;; +----------------------------------------------------------------------------------------------------------------+
 
-(defn format-personal-collection-name
-  "Constructs the personal collection name from user name."
-  [first-name last-name]
-  (tru "{0} {1}''s Personal Collection" first-name last-name))
+(s/defn format-personal-collection-name :- su/NonBlankString
+  "Constructs the personal collection name from user name.
+  When displaying to users we'll tranlsate it to user's locale,
+  but to keeps things consistent in the database, we'll store the name in site's locale.
+
+  Practically, use `user-or-site` = `:site` when insert or update the name in database,
+  and `:user` when we need the name for displaying purposes"
+  [first-name last-name user-or-site]
+  (assert (#{:user :site} user-or-site))
+  (if (= :user user-or-site)
+    (tru "{0} {1}''s Personal Collection" first-name last-name)
+    (trs "{0} {1}''s Personal Collection" first-name last-name)))
```

Comment:
If very interested in knowing if we could do this using macro? something like `((if pred tru trs) "string {0}" var-0))`

Reply:
No, because macros are expanded prior to compilation, since they transform the source code that is ultimately compiled. They're not a drop-in replacement for normal functions. What you have now is fine, I think.